lib/rubocop/cop/magic_numbers/no_assignment.rb

```diff
@@ -72,39 +63,26 @@ def on_multiple_assign(node)
 
         private
 
-        def illegal_scalar_types
-          ILLEGAL_SCALAR_TYPES[cop_config['ForbiddenNumerics'] || 'All']
-        end
-
-        def illegal_scalar_types_ast
-          "{#{illegal_scalar_types.join(' ')}}"
-        end
-
-        def magic_number_argument_pattern
-          MAGIC_NUMBER_ARGUMENT_PATTERN.call(illegal_scalar_types_ast:)
-        end
-
-        def magic_number_multi_assign_pattern
-          MAGIC_NUMBER_MULTI_ASSIGN_PATTERN.call(illegal_scalar_types_ast:)
-        end
-
-        def assignment?(node)
-          # Only match on method names that resemble assignments
-          method_name(node).end_with?('=')
-        end
-
-        def unary?(node)
-          # Only match on method names that are unary invocations, so 1 character
-          # long
-          method_name(node).length == UNARY_LENGTH
-        end
+        def illegal_scalar_argument_to_setter?(node)
+          method = node_matches_pattern?(
```

Comment:
This is a bit dodgy because we're relying on the capture of the method from the pattern coming back, but it's the best way I could manage to get access to it; https://docs.rubocop.org/rubocop-ast/node_pattern.html#to-call-functions says we should be able to invoke this function in the pattern matcher directly, but I kept getting a parser error on that pattern, so this is what I've gone with.